src/main/java/nl/ordina/jobcrawler/controller/MylerVacancyScraper.java

```diff
@@ -21,42 +20,33 @@ At this moment getVacancies() returns a list with Vacancies that only exist of a
 @Component
 public class MylerVacancyScraper extends VacancyScraper {
 
-    private static final String BROKER = "Myler";
-    private static final String SEARCH_URL = "https://www.myler.nl/opdrachten/?search=" + SEARCH_TERM;
-
-    private List<Vacancy> vacancies = new ArrayList<>();
-
-    private final ConnectionDocumentService connectionDocumentService;
-
-    @Autowired
     public MylerVacancyScraper(ConnectionDocumentService connectionDocumentService) {
-        this.connectionDocumentService = connectionDocumentService;
+        super(connectionDocumentService);
+        BROKER = "Myler";
+        SEARCH_URL = "https://www.myler.nl/opdrachten/?search=" + SEARCH_TERM;
     }
 
     @Override
-    public List<Vacancy> getVacancies() throws IOException {
-        /*
-        getVacancies connect to the SEARCH_URL and requests this html page.
-        If the response is not null it selects the a tags and requests the urls to the vacancy pages.
-        Currently this scraper only stores the url and broker into a 'vacancy' object and returns this in a list.
-         */
-        Document mylerVacancies = connectionDocumentService.getConnection(SEARCH_URL);
-        if(mylerVacancies != null) {
-            Elements vacancyURLs = mylerVacancies.getElementsByClass("headfirst-plugin-item").select("a");
-            for(Element vacancyURL : vacancyURLs) {
-                Vacancy vacancy = new Vacancy();
-                String url = vacancyURL.attr("href");
-                vacancy.setVacancyURL(url);
-                vacancy.setBroker(BROKER);
-                this.vacancies.add(vacancy);
+    protected List<VacancyURLs> getVacancyURLs() throws IOException {
+        //  Returns a List with VacancyURLs
+        List<VacancyURLs> vacancyURLs = new ArrayList<>();
+        int totalNumberOfPages = 1;
+        for(int pageNumber = 1; pageNumber <= totalNumberOfPages; pageNumber++) {
```

Comment:
What is `pageNumber ` used for exactly?

Reply:
Yacht currently displays a maximum of 15 vacancies per page. As we start scraping the first page anyway, we'd like to know if more pages are available in order to get all vacancies. This implementation is copied from the Arabot project and adjusted so that it works at this moment. Might need to discuss if this is the right way to do this.